### ISSUE TITLE
cache with Redis 

### ISSUE THREAD

**Issue by Alexis90** [arthurfiorette/axios-cache-interceptor#517]
### Describe the bug

When using Redis storage, I got a TypeError
```
TypeError: Cannot read properties of undefined (reading 'id')
    at o (D:\dev\axios-cache\node_modules\axios-cache-interceptor\dist\index.cjs:2:5515)
```

### To Reproduce

code repo: https://github.com/Alexis90/axios-cache-debug.git

```
const client = redis.createClient({
  url: process.env.REDIS_URL,
});


const redisStorage = buildStorage({
  async find(key) {
    const result = await client.get(`axios-cache:${key}`);
    return JSON.parse(result);
  },

  async set(key, value) {
    await client.set(`axios-cache:${key}`, JSON.stringify(value), {
      PXAT: canStale(value) ? value.expiresAt : undefined,
    });
  },

  async remove(key) {
    await client.del(`axios-cache:${key}`);
  },
});

setupCache(axios, {
  storage: redisStorage,
});

app.get("/api/ditto", async (req, res) => {
  const result = await axios.get("https://pokeapi.co/api/v2/pokemon/ditto");
  res.status(200).json(result.data);
});
```

### Expected behavior

Set the request response into Redis storage as a cache automatically

### Additional context

- Axios: `1.3.4`
- Axios Cache Interceptor: `1.0.1`
- What storage is being used: `Redis`
- Node Version: `18`
- Browser (if applicable): `Chrome`

**Comment by arthurfiorette** [comment]
Thanks! I'm out of town today, but I'll look into it as soon as possible.

**Comment by nlamarti** [comment]
`canStale(value)` is the issue here as in the case of a `new` or `loading` cache state `value.data` is undefined but is used in the canStale:
```
export function canStale(value: CachedStorageValue): boolean {
  const headers = value.data.headers;

  return (
    Header.ETag in headers ||
    Header.LastModified in headers ||
    Header.XAxiosCacheEtag in headers ||
    Header.XAxiosCacheStaleIfError in headers ||
    Header.XAxiosCacheLastModified in headers
  );
}
```
using the latest version of both redis and axios-cache-interceptor the redis store example should be:

```
const client = redis.createClient({
    url: process.env.REDIS_URL
});

const redisStorage = buildStorage({
    async find(key) {
        if (!client.isOpen) await client.connect();
        const result = await client.get(`axios-cache:${key}`);
        return JSON.parse(result);
    },

    async set(key, value) {
        if (!client.isOpen) await client.connect();
        await client.set(`axios-cache:${key}`, JSON.stringify(value), {
            PXAT: (value.data !== undefined && canStale(value)) ? value.createdAt + value.ttl : undefined,
        });
    },

    async remove(key) {
        if (!client.isOpen) await client.connect();
        await client.del(`axios-cache:${key}`);
    },
});

setupCache(axios, {
    storage: redisStorage,
});

```

**Comment by arthurfiorette** [comment]
Hmm, seems out that any error thrown by an adapter, `TypeError: Cannot read properties of undefined (reading 'id')` is thrown because the response interceptor is not able to handle it. I'll add a try catch to help further errors.

**Comment by arthurfiorette** [comment]
About the error itself, it was a problem when `canStale` was called with a non-cache storage value, which was fixed by #514.
